DashAI/back/dependencies/database/models.py

```diff
@@ -99,3 +100,39 @@ def set_status_as_finished(self) -> None:
     def set_status_as_error(self) -> None:
         """Update the status of the run to error."""
         self.status = RunStatus.ERROR
+
+
+class Plugin(Base):
+    __tablename__ = "plugin"
+    """
+    Table to store all the information about a dataset.
+    """
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(String, unique=True, nullable=False)
+    author: Mapped[str] = mapped_column(String, nullable=False)
+    tags: Mapped[List["Tag"]] = relationship(
+        back_populates="plugin", cascade="all, delete", lazy="selectin"
+    )
+    status: Mapped[Enum] = mapped_column(
+        Enum(PluginStatus), nullable=False, default=PluginStatus.REGISTERED
+    )
+    summary: Mapped[str] = mapped_column(String, nullable=False)
+    description: Mapped[str] = mapped_column(String, nullable=False)
+    description_content_type: Mapped[str] = mapped_column(String, nullable=False)
+    created: Mapped[DateTime] = mapped_column(DateTime, default=datetime.now)
+    last_modified: Mapped[DateTime] = mapped_column(
+        DateTime,
+        default=datetime.now,
+        onupdate=datetime.now,
+    )
+
+
+class Tag(Base):
+    __tablename__ = "tag"
+    """
+    Table to store all the tags related to a plugin
+    """
+    id: Mapped[int] = mapped_column(primary_key=True)
+    plugin: Mapped["Plugin"] = relationship(back_populates="tags")
+    plugin_id: Mapped[int] = mapped_column(ForeignKey("plugin.id"))
+    name: Mapped[Enum] = mapped_column(Enum(PluginTag), nullable=False)
```

Comment:
This complicates me a little. What will happen if we dynamically add more plugins, for example, when a new model type is installed that is taged with something that did not exist in the original enum? 
I think we should not have an enum at all, but just leave this field open.

So we could also have tags like Scikit, HuggingFace, SHAPLike or others.

Reply:
At the beginning I proposed to use an enum to avoid the creation of several same tags but with typographical errors, in order to force the plugin developer to check that the declared plugins are well typed. 
But now I think this is too restrictive, at this stage I think it is better to be more flexible, I will create a task in notion to think about it in the future.